src/hotspot/os/linux/os_linux.cpp

```diff
@@ -672,7 +672,8 @@ static void *thread_native_entry(Thread *thread) {
   // and we did not see any degradation in performance without `alloca()`.
   static int counter = 0;
   int pid = os::current_process_id();
-  void *stackmem = alloca(((pid ^ counter++) & 7) * 128);
+  int random = ((pid ^ counter++) & 7) * 128;
+  void *stackmem = alloca(random != 0 ? random : 1); // ensure we allocate > 0
```

Comment:
Is it better to use `random > 0` rather than `!= 0`?
Both `pid` and `couter` are `int`, so I concern MSB in `random` might be `1` the result of XOR. (it is corner case I think...)